### ISSUE TITLE
Use sphinxcontrib apidoc for autogenerated API docs

### ISSUE THREAD

**Issue by djhoese** [pangeo-data/xbatcher#53]
I've had a lot of luck using [sphinxcontrib-apidoc](https://pypi.org/project/sphinxcontrib-apidoc/) to automatically generate the API `.rst` files when the sphinx documentation is rendered and it has saved me a lot of time. I don't know if any other pangeo projects use it, but I'd be willing to add it here if no one has a problem with that.

**Comment by maxrjones** [comment]
Hi @djhoese, sorry that this issue went without a response for so long! Documentation improvements are always appreciated - if you still think this would save time you are welcome to open a PR.

**Comment by djhoese** [comment]
Yeah, I think this tool is always pretty helpful. Given my involvement with the Pangeo ML/NASA project I should be able to find time this week to tackle this. I'll add it to my TODO list. Thanks for responding to old issues!
